app/uk/gov/hmrc/rasapi/repository/GridFsTTLIndexing.scala

```diff
@@ -34,55 +33,46 @@ trait GridFsTTLIndexing {
   private lazy val OptExpireAfterSeconds = "expireAfterSeconds"
   protected lazy val UploadDate = "uploadDate"
 
-  def addAllTTLs(gfs: GridFS[BSONSerializationPack.type])(implicit ec: ExecutionContext): Future[Seq[Boolean]] = {
-    Future.sequence(Seq(
-      addFilesTTL(gfs),
-      addChunksTTL(gfs)
-    ))
+  private def listAllIndexes(mdb: MongoDatabase, collectionName: String): Seq[Document] = {
+    log.info(s"[GridFsTTLIndexing][retrieveLastUpdatedIndex] Searching $collectionName for all indexes.")
+    Await.result(mdb.getCollection(collectionName)
+      .listIndexes().toFuture(), Duration(10, SECONDS))
   }
 
-  def addChunksTTL(gfs: GridFS[BSONSerializationPack.type])(implicit ec: ExecutionContext): Future[Boolean] = {
-    addTTL(gfs.chunks)
+  private def retrieveIndex(mdb: MongoDatabase, collectionName: String, indexName: String): Seq[Document] = {
+    log.info(s"[GridFsTTLIndexing][retrieveLastUpdatedIndex] Searching $collectionName for $LastUpdatedIndex")
+    Await.result(mdb.getCollection(collectionName)
+      .listIndexes()
+      .filter(index =>
+    index.containsValue(indexName)
+    ).toFuture(), Duration(10, SECONDS))
   }
 
-  def addFilesTTL(gfs: GridFS[BSONSerializationPack.type])(implicit ec: ExecutionContext): Future[Boolean] = {
-    addTTL(gfs.files)
+  private def dropIndex(mdb: MongoDatabase, collectionName: String, indexName: String): Void = {
+    log.info(s"[GridFsTTLIndexing][checkAndEnsureTTL] Dropping the existing $indexName")
+    Await.result(mdb.getCollection(collectionName).dropIndex(indexName).toFuture(), Duration(10, SECONDS))
   }
 
-  private def addTTL(collection: GenericCollection[BSONSerializationPack.type])(implicit ec: ExecutionContext): Future[Boolean] = {
-    import reactivemongo.bson.DefaultBSONHandlers._
-    val indexes = collection.indexesManager.list()
-    indexes.flatMap {
-      idxs => {
-
-        val idxToUpdate = idxs.find(index =>
-          index.eventualName == LastUpdatedIndex
-            && index.options.getAs[BSONLong](OptExpireAfterSeconds).getOrElse(BSONLong(expireAfterSeconds)).as[Long] != expireAfterSeconds)
-
-        if (idxToUpdate.isDefined) {
-          for {
-            _ <- collection.indexesManager.drop(idxToUpdate.get.eventualName)
-
-            updated <- ensureLastUpdated(collection)
-          } yield updated
-        }
-        else {
-          ensureLastUpdated(collection)
-        }
-      }
-    }
-    log.info(s"[GridFsTTLIndexing][addTTL] Creating time to live for entries in ${collection.name} to $expireAfterSeconds seconds")
-    ensureLastUpdated(collection)
+  private def createIndex(mdb: MongoDatabase, collectionName: String, indexName: String): String = {
```

Comment:
this count also return a Future for simplicity - if this needs to happen prior to something else down the line, it would be a good idea to add an `andThen` or some such at that point